sdk/core/azure-core/test/ut/environmentLogLevelListener_test.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifdef __APPLE__
+#define PUTENV(x) putenv((char*)x)
+#else
+#define PUTENV(x) _putenv(std::string(x).c_str());
+#endif // __APPLE__
```

Comment:
I'd make a function inside an unnamed namespace, using macro is the last resort.
Here is cross-platform `setenv()`: https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/identity/azure-identity/test/ut/environment_credential_test.cpp#L12-L47

Reply:
good point,